docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+
+The most important effect of this change is the definition of a new semantic for the Machine's “Ready” condition, that
+will now clearly represent the "machine can host workloads" (prior art Kubernetes nodes are ready when "node can host pods"). 
+To improve the benefit of this change:
+
+- This proposal is ensuring that whenever Machine ready is used, it always means the same thing (e.g. replica counters)
+- This proposal is also changing contract fields where ready was used improperly to represent 
+  initial provisioning (k8s API conventions suggest to use ready only for long-running process).
+
+All in all, Machine's Ready should be much more clear, consistent, intuitive after proposed changes.
+But there is more.
+
+This proposal is also dropping Ready condition from higher level abstractions in Cluster API.
+
+Instead, where not already present, this proposal is introducing a new Available condition that better represents
+the fact that those objects are operational even if there is a certain degree of not readiness in the system
+or if lifecycle operations are happening (prior art Available condition in K8s Deployments).
+
+Last but not least:
+
+- With the changes to the semantic of Ready and Available conditions, it is now possible to add conditions about
+  surfacing that lifecycle operations are happening, e.g. scaling up.
+- As suggested by K8s API conventions, this proposal is also making sure all conditions are consistent and have
+  uniform meaning across all resource types
+- Additionally, we are enforcing the same consistency for replica counters and other status fields.
+
+### Transition to K8s API conventions aligned conditions
+
+K8s is undergoing an effort of standardizing usage of conditions across all resource types, and the transition to
+the v1beta2 API version is a great opportunity for Cluster API to align to this effort.
+
+The value of this transition is substantial, because the differences that exists today's are really confusing for users;
+those differences are also making it harder for ecosystem tools to build on top of Cluster API, and in some cases
+even confusing new (and old) contributors.
+
+With this proposal Cluster API will close the gap with K8s API conventions in regard to:
+- Polarity: Condition type names should make sense for humans; neither positive nor negative polarity can be recommended
+  as a general rule (already implemented by [#10550](https://github.com/kubernetes-sigs/cluster-api/pull/10550))
+- Use of the Reason field is required (currently in Cluster API reasons is added only when condition are false)
+- Controllers should apply their conditions to a resource the first time they visit the resource, even if the status is Unknown.
+  (currently Cluster API controllers add conditions at different stages of the reconcile loops)
+- Cluster API is also dropping its own Condition type and start using metav1.Conditions from the Kubernetes API.
```

Comment:
Is there a specific min Kubernetes version we're targeting?

Reply:
Not really, 
this depends only by the Kubernetes version we are using in CAPI go.mod file, and we are already at a version that has all the types we need to have

Reply:
If I see correctly metav1.Conditions are unchanged since 4 years: https://github.com/kubernetes/kubernetes/blame/3f7d4b787b74a2d110901d333f83e99c0e5e49c9/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1532

(not sure why you're asking though :))

Reply:
Resolving, I assume this is clear now